src/test/java/bio/terra/service/auth/iam/sam/SamApiServiceTest.java

```diff
@@ -2,63 +2,81 @@
   private SamApiService samApiService;
+  private Map<String, ApiClient> unauthorizedApiClients;
+  private Map<String, ApiClient> authorizedApiClients;
 
   private static final String TOKEN = "some-access-token";
   private static final int OPERATION_TIMEOUT_SECONDS = 123;
   private static final int OPERATION_TIMEOUT_MILLIS = OPERATION_TIMEOUT_SECONDS * 1000;
 
-  @Before
-  public void setUp() throws Exception {
-    samApiService = new SamApiService(samConfig, configurationService);
-
+  @BeforeEach
+  void setUp() {
     when(configurationService.getParameterValue(ConfigEnum.SAM_OPERATION_TIMEOUT_SECONDS))
         .thenReturn(OPERATION_TIMEOUT_SECONDS);
-  }
-
-  @Test
-  public void testSamApiClientReadTimeouts() {
-    // We only have one unauthorized Sam API -- StatusApi -- and this is unlikely to change.
-    assertThat(
-        "StatusApi keeps default read timeout",
-        samApiService.statusApi().getApiClient().getReadTimeout(),
-        equalTo(new ApiClient().getReadTimeout()));
-
-    var authorizedApiClients =
+    samApiService = new SamApiService(samConfig, configurationService);
+    unauthorizedApiClients = Map.of("StatusApi", samApiService.statusApi().getApiClient());
+    authorizedApiClients =
         Map.of(
             "ResourcesApi", samApiService.resourcesApi(TOKEN).getApiClient(),
             "GoogleApi", samApiService.googleApi(TOKEN).getApiClient(),
             "UsersApi", samApiService.usersApi(TOKEN).getApiClient(),
             "TermsOfServiceApi", samApiService.termsOfServiceApi(TOKEN).getApiClient(),
             "GroupApi", samApiService.groupApi(TOKEN).getApiClient());
+  }
+
+  @Test
+  void testSamApiClientReadTimeouts() {
+    unauthorizedApiClients.forEach(
+        (name, client) ->
+            assertThat(
+                name + " keeps default read timeout",
+                client.getReadTimeout(),
+                equalTo(new ApiClient().getReadTimeout())));
     authorizedApiClients.forEach(
         (name, client) ->
             assertThat(
                 name + " has expected read timeout",
                 client.getReadTimeout(),
                 equalTo(OPERATION_TIMEOUT_MILLIS)));
   }
+
+  @Test
+  void testSamApiClientsShareHttpClient() {
+    List<ApiClient> apiClients = new ArrayList<>();
+    apiClients.addAll(unauthorizedApiClients.values());
+    apiClients.addAll(authorizedApiClients.values());
+
+    // Our intention is that all Sam ApiClients share a common HttpClient, but under the covers
+    // we could have many HttpClients built to support different configuration that all reuse the
+    // same connection pool.
+    // As a proxy check, we make sure that the same connection pool is reused.
+    List<ConnectionPool> connectionPools =
+        apiClients.stream().map(ac -> ac.getHttpClient().connectionPool()).distinct().toList();
+    assertThat("All Sam ApiClients share a connection pool", connectionPools, hasSize(1));
```

Comment:
minor suggestion, you could use `count()` here instead of `toList() ... hasSize(1)`
```suggestion
    int numConnectionPools =
        apiClients.stream().map(ac -> ac.getHttpClient().connectionPool()).distinct().count();
    assertThat("All Sam ApiClients share a connection pool", numConnectionPools, is(1));
```